lib/bumblebee/text.ex

```diff
@@ -305,6 +305,71 @@ defmodule Bumblebee.Text do
   defdelegate text_classification(model_info, tokenizer, opts \\ []),
     to: Bumblebee.Text.TextClassification
 
+  @type text_embedding_input :: String.t()
+  @type text_embedding_output :: %{embedding: Nx.Tensor.t()}
+
+  @doc """
+  Builds serving for text embeddings.
+
+  The serving accepts `t:text_embedding_input/0` and returns
+  `t:text_embedding_output/0`. A list of inputs is also supported.
+
+  ## Options
+    * `:output_attribute` - the attribute of the model output map to
+      retrieve. When the output is a single tensor (rather than a map),
+      this option is ignored. Defaults to `:pooled_state`
+
+    * `:output_pool` - pooling to apply on top of the model output, in case
+      it is not already a pooled embedding. Supported values: `nil` and
+      `:mean`. Defaults to `nil`, which applies no pooling on top of the
+      model output.
+
+    * `:embedding_processor` - a post-processing step to apply to the
+      embedding. Supported values: `nil` and `:l2_norm`. Defaults to
+      `nil`, which applies no post-processing step to the embedding.
```

Comment:
We usually don't list `nil` explicitly, it's always the case for optional options :)

```suggestion
    * `:output_pool` - pooling to apply on top of the model output, in case
      it is not already a pooled embedding. Supported values: `:mean`. By
      default no pooling is applied

    * `:embedding_processor` - a post-processing step to apply to the
      embedding. Supported values: `:l2_norm`. By default the output is
      returned as is
```